The War Within - Update

DonnieDice opened the pull request:
### Changelog

#### Version 2.0.0-------------------------------------------------------------------

- Fixed - Resolved an issue where `TooltipUtil.SurfaceArgs` was called without proper checks, leading to Lua errors
- Fixed - Corrected quest tracking when using the `QUEST_ACCEPTED`, `QUEST_REMOVED`, and `QUEST_WATCH_LIST_CHANGED` events
- Added - Fallback logic for tooltip processing in `GetQuestProgress` to handle cases where `TooltipUtil.SurfaceArgs` is unavailable
- Updated - Enhanced `UpdateQuestIcon` function to better handle quest progress updates and ensure icons display correctly for all quest states
- Updated - Refined event handling for `PLAYER_LOGIN` to correctly populate the `ActiveWorldQuests` table based on the player's current map
- Improved - Re-anchored quest icons on nameplates during the `ADDON_LOADED` event to ensure settings are correctly applied
- Updated - ToC for _retail_ - The War Within Expansion Release Patch (10.2.0.56110) [ToC.110002]

---

DonnieDice committed:
The War Within

Update for WW

DonnieDice committed:
v2.0.0

- Fixed - Resolved an issue where `TooltipUtil.SurfaceArgs` was called without proper checks, leading to Lua errors
- Fixed - Corrected quest tracking when using the `QUEST_ACCEPTED`, `QUEST_REMOVED`, and `QUEST_WATCH_LIST_CHANGED` events
- Added - Fallback logic for tooltip processing in `GetQuestProgress` to handle cases where `TooltipUtil.SurfaceArgs` is unavailable
- Updated - Enhanced `UpdateQuestIcon` function to better handle quest progress updates and ensure icons display correctly for all quest states
- Updated - Refined event handling for `PLAYER_LOGIN` to correctly populate the `ActiveWorldQuests` table based on the player's current map
- Improved - Re-anchored quest icons on nameplates during the `ADDON_LOADED` event to ensure settings are correctly applied
- Updated - ToC for _retail_ - The War Within Expansion Release Patch (10.2.0.56110) [ToC.110002]

zcourge commented:
Awesome work, thanks for this PR

Numynum commented:
Fyi, `TooltipUtil.SurfaceArgs` has done nothing for ages, and has been permanently deleted in 11.0
no point in checking if it exists, the code related to it can just be deleted

DonnieDice committed:
v11.0.2

- Fixed - Resolved an issue where `TooltipUtil.SurfaceArgs` was called without proper checks, leading to Lua errors
- Fixed - Corrected quest tracking when using the `QUEST_ACCEPTED`, `QUEST_REMOVED`, and `QUEST_WATCH_LIST_CHANGED` events
- Added - Fallback logic for tooltip processing in `GetQuestProgress` to handle cases where `TooltipUtil.SurfaceArgs` is unavailable
- Updated - Enhanced `UpdateQuestIcon` function to better handle quest progress updates and ensure icons display correctly for all quest states
- Updated - Refined event handling for `PLAYER_LOGIN` to correctly populate the `ActiveWorldQuests` table based on the player's current map
- Improved - Re-anchored quest icons on nameplates during the `ADDON_LOADED` event to ensure settings are correctly applied
- Updated - ToC for _retail_ - The War Within Expansion Release Patch (10.2.0.56110) [ToC.110002]

DonnieDice committed:
v11.0.2

- Fixed - Resolved an issue where `TooltipUtil.SurfaceArgs` was called without proper checks, leading to Lua errors
- Fixed - Corrected quest tracking when using the `QUEST_ACCEPTED`, `QUEST_REMOVED`, and `QUEST_WATCH_LIST_CHANGED` events
- Added - Fallback logic for tooltip processing in `GetQuestProgress` to handle cases where `TooltipUtil.SurfaceArgs` is unavailable
- Updated - Enhanced `UpdateQuestIcon` function to better handle quest progress updates and ensure icons display correctly for all quest states
- Updated - Refined event handling for `PLAYER_LOGIN` to correctly populate the `ActiveWorldQuests` table based on the player's current map
- Improved - Re-anchored quest icons on nameplates during the `ADDON_LOADED` event to ensure settings are correctly applied
- Updated - ToC for _retail_ - The War Within Expansion Release Patch (10.2.0.56110) [ToC.110002]

DonnieDice committed:
v5.4.1

- Added - [release.yml] - [/.github/workflows/]
- Added - [README.md]
- Updated - Project information - [QuestPlates.toc]
- Fixed - Version Numbering - [QuestPlates.toc]

DonnieDice committed:
v5.4.2

- Added - New Folder - [/docs]
- Added - changelog.txt - [/docs]
- Moved - [CHANGES.md] - [/docs]

DonnieDice committed:
v5.4.3

- Added - [.pkgmeta]

DonnieDice committed:
v5.4.4

- Fixed - Package as - [.pkgmeta]

DonnieDice committed:
v5.4.5

- Fixed - Package as - [.pkgmeta]

DonnieDice committed:
v5.4.6

- Updated - ToC for _retail_ - [ToC.110007]

DonnieDice committed:
v5.4.7

- Updated - Added note about Semlar - [README.md]
- Updated - Added color to AddOn Menu Title - [QuestPlates.toc]

DonnieDice committed:
Update README.md

Oops

DonnieDice committed:
v5.4.8

- Updated - Retail TOC to reflect version 110100 [QuestPlates.toc]
- Updated - Retail BLU version number to v5.4.8 in TOC [QuestPlates.toc]

DonnieDice committed:
v5.4.9

- Added - Login message displaying addon name, version, and icon on player login, styled similarly to FFLU - [QuestPlates.lua]
- Fixed - Resolved a case sensitivity issue in `PLAYER_LOGIN` where `task.questId` was used instead of `task.questID`, causing nil `questID` errors in certain zones (e.g., Undermine, Nazjatar) - [QuestPlates.lua]
- Added - Nil check for `questID` in `PLAYER_LOGIN` to prevent invalid API calls to `C_TaskQuest.GetQuestInfoByQuestID` - [QuestPlates.lua]
- Updated - Styled TOC title with custom color #58be81 and added email field, aligning with FFLU addon style - [QuestPlates.toc]

DonnieDice committed:
v5.4.10

- Updated - TOC version bump [QuestPlates.toc]
- Updated - Email address updated in TOC [QuestPlates.toc]

DonnieDice committed:
docs: update README to reflect project continuation

DonnieDice committed:
docs: update README to reflect deprecation and new project

DonnieDice committed:
ci: quote release-type key in release.yml for YAML linting
